packages/ui5-application-inquirer/src/prompts/prompts1.ts

```diff
@@ -281,10 +281,21 @@ export function getAddFlpConfigPrompt(
 /**
  * Get the `enableVirtualEndpoints` prompt.
  *
+ * @param isCapProject is this a CAP project - additional checks performed if true
+ * @param capCdsInfo CDS UI5 plugin information
  * @returns the `enableVirtualEndpoints` prompt
  */
-export function getEnableVirtualEndpoints(): UI5ApplicationQuestion {
+export function getEnableVirtualEndpoints(
+    isCapProject: boolean,
+    capCdsInfo?: CdsUi5PluginInfo
+): UI5ApplicationQuestion {
     return {
+        when: (answers: UI5ApplicationAnswers): boolean => {
+            if (isCapProject) {
+                return capCdsInfo?.isCdsUi5PluginEnabled || !!answers.enableTypeScript;
```

Comment:
Should l we be checking for `enableTypescript` as this implies knowledge about other flows and tightly couples to the typescript flow. I had previously removed `enableTypescript` checks and rely on the cds info only.

Reply:
As discussed, this needs following up as the writers don't currently support the assumption that cds ui5 plugin will be enabled if the mid cds version is met